### ISSUE TITLE
[23.1] debugSymbolsQuarkus and debugSymbolsSmokeGDB tests fail due to experimental options warnings

### ISSUE THREAD

**Issue by jerboaa** [Karm/mandrel-integration-tests#188]
They fail with:

```
Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 271.835 s <<< FAILURE! - in org.graalvm.tests.integration.DebugSymbolsTest
debugSymbolsQuarkus{TestInfo}  Time elapsed: 234.044 s  <<< FAILURE!
org.opentest4j.AssertionFailedError:
build-and-run.log log should not contain error or warning lines that are not whitelisted. See /disk/graal/upstream-sources/mandrel-integration-tests/testsuite/target/archived-logs/org.graalvm.tests.integration.DebugSymbolsTest/debugSymbolsQuarkus/build-and-run.log and check these offending lines:
Warning: The option '-H:+TrackNodeSourcePosition' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:ReflectionConfigurationResources=META-INF/native-image/io.netty/netty-transport/reflection-config.json' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:+DebugCodeInfoUseSourceMappings' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: Please re-evaluate whether any experimental option is required, and either remove or unlock it. The build output lists all active experimental options, including where they come from and possible alternatives. If you think an experimental option should be considered as stable, please file an issue.
Warning: The option '-H:DebugInfoSourceSearchPath=app-sources' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:Log=registerResource:' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future. ==> expected: <true> but was: <false>
        at org.graalvm.tests.integration.DebugSymbolsTest.debugSymbolsQuarkus(DebugSymbolsTest.java:276)
```

and:

```
debugSymbolsSmokeGDB{TestInfo}  Time elapsed: 37.451 s  <<< FAILURE!
org.opentest4j.AssertionFailedError:
build-and-run.log log should not contain error or warning lines that are not whitelisted. See /disk/graal/upstream-sources/mandrel-integration-tests/testsuite/target/archived-logs/org.graalvm.tests.integration.DebugSymbolsTest/debugSymbolsSmokeGDB/build-and-run.log and check these offending lines:
Warning: The option '-H:+PreserveFramePointer' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:-DeleteLocalSymbols' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:+TrackNodeSourcePosition' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:GenerateDebugInfo=1' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:+OmitInlinedMethodDebugLineInfo' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: The option '-H:+DebugCodeInfoUseSourceMappings' is experimental and must be enabled via '-H:+UnlockExperimentalVMOptions' in the future.
Warning: Please re-evaluate whether any experimental option is required, and either remove or unlock it. The build output lists all active experimental options, including where they come from and possible alternatives. If you think an experimental option should be considered as stable, please file an issue. ==> expected: <true> but was: <false>
        at org.graalvm.tests.integration.DebugSymbolsTest.debugSymbolsSmokeGDB(DebugSymbolsTest.java:190)
```

**Comment by zakkak** [comment]
As discussed in https://github.com/oracle/graal/issues/7356 these options will remain experimental so we need to wrap them in `UnlockExperimentalVMOptions`

**Comment by jerboaa** [comment]
Remaining issues for these tests are down to [netty's native-image config](https://github.com/netty/netty/pull/13596) as far as I can see, which is hardly something we control. This is being tracked in netty. Can we allow those log lines for now in order to reduce test noise?

`debugSymbolsQuarkus`:
https://github.com/graalvm/mandrel/actions/runs/6159630722/job/16716434614#step:11:10310

**Comment by zakkak** [comment]
> Can we allow those log lines for now in order to reduce test noise?

+1 from me

**Comment by jerboaa** [comment]
Remaining issues should be fixed with #203
